ui/core/src/model/log-data.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright 2025 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Notice } from './notice';
+import { AbsoluteTimeRange } from './time';
+import { Labels } from './time-series-queries';
+
+export interface LogEntry {
+  timestamp: number;
+  line: string;
+  labels: Labels;
+}
+
+export interface LogData {
+  timeRange?: AbsoluteTimeRange;
+  entries: LogEntry[];
+  metadata?: LogMetadata;
+  totalCount: number;
```

Comment:
not sure if all the clients will provide this value by default, in some cases might be an estimation or it will require a new query. Maybe we can make it optional

```suggestion
  totalCount?: number;
```